client/actions/authActions.js

```diff
@@ -2,49 +2,88 @@ import axios from 'axios';
 import setAuthToken from '../utils/setAuthToken';
 import jwtDecode from 'jwt-decode';
 import { USER_AUTHENTICATED, RESET_PASSWORD_SUCCESS } from './types';
-
+/*global Materialize */
+/**
+ * 
+ * 
+ * @export
+ * @param {any} user 
+ * @returns {void}
+ */
 export function setCurrentUser(user) {
     return {
         type: USER_AUTHENTICATED,
         user
     };
 }
+/**
+ * 
+ * 
+ * @export
+ * @param {any} data 
+ * @returns {void}
+ */
 export function login(data) {
     return dispatch => {
-        return axios.post('/api/user/signin', data).then(res => {
+        return axios.post('/api/v1/user/signin', data).then(res => {
             const token = res.data.token;
             localStorage.setItem('jwtToken', token);
             setAuthToken(token);
             dispatch(setCurrentUser(decode(token)));
         });
     };
 }
+/**
+ * 
+ * 
+ * @export
+ * @param {any} userData 
```

Comment:
I kinda disagree with you that you don't know what most of those params are. even me from mere looking I know is an object. Consider refactoring this to use appropriate value